### Background Information
As a business analyst, I found a strong positive correlation between the sales of two different products: BobaWonder (bubble tea) and WaterCure (premium water). This result puzzled the company, as the two products cater to different customer bases. I suspect that the high sales of both products could be linked to external factors, like the warm Australian climate. To explore this further, I am provided with daily sales data for WaterCure and daily temperature data from January to August 2024, with the goal of determining whether temperature influences product sales.

In [4]:
# identify the data type and what you wanna do ?

### 1. Get a summary of the daily sales and daily temperature (i.e., the count, mean, standard deviation, minimum, maximum, ... etc. ) and interpret the summary in the Markdown or Text Cell (e.g., what does maximum mean for the daily temperature field?) after the code

In [17]:
import pandas as pd



In [32]:
ls

Hsieh_L_1.ipynb                    WaterCure_analysis_Hsieh_Luna.csv
Hsieh_L_2.ipynb                    daily_sales.csv
Hsieh_L_3.ipynb                    hw1-q.pptx


In [34]:
df= pd.read_csv ('daily_sales.csv',skiprows = 2)

In [36]:
df.head()

Date  Daily Units Sold  Daily Unit Price  Daily Temperature (C)
0  1/1/2024                91              24.0                     25
1  1/2/2024                90              24.0                     24
2  1/3/2024                70              24.0                     19
3  1/4/2024                89              24.0                     23
4  1/5/2024               100              24.0                     36

In [37]:
df.tail()

Date  Daily Units Sold  Daily Unit Price  Daily Temperature (C)
239  8/27/2024                29              22.2                     16
240  8/28/2024                59              22.2                     21
241  8/29/2024                33              22.2                     17
242  8/30/2024                33              22.2                     17
243  8/31/2024                37              22.2                     18

In [38]:
df.describe()

Daily Units Sold  Daily Unit Price  Daily Temperature (C)
count        244.000000        244.000000             244.000000
mean          66.479508         21.894262              19.512295
std           19.003021          1.223084               5.374442
min           11.000000         20.500000               7.000000
25%           57.000000         20.875000              16.000000
50%           63.500000         21.750000              19.000000
75%           81.250000         22.200000              23.000000
max          100.000000         24.000000              36.000000

### Summary of Daily Sales and Temperature Data

#### 1. Daily Units Sold:
- Count: Data for 244 days.
- Mean (Average): About 66 units sold per day on average.
- Standard Deviation: Sales vary by around 19 units daily.
- Minimum: The lowest number of units sold in a day was 11.
- Maximum: The highest number of units sold in a day was 100.
- Percentiles:
  - 25% of the days had sales of 57 units or fewer.
  - Half the days (median) had sales of 63.5 units.
  - 75% of the days had sales of 81 units or fewer.

#### 2. Daily Unit Price:
- Mean: The average price per unit was about 21.89.
- Standard Deviation: Prices varied slightly by about 1.22.
- Minimum: The lowest price was 20.5.
- Maximum: The highest price was 24.
- Percentiles:
  - 25% of the days had prices below 20.875.
  - The median price was 21.75.
  - 75% of the days had prices below 22.2.

#### 3. Daily Temperature (°C):
- Mean: The average temperature was about 19.5°C.
- Standard Deviation: Temperature fluctuated by around 5.37°C.
- Minimum: The coldest day was 7°C.
- Maximum: The hottest day was 36°C.
- Percentiles:
  - 25% of the days were cooler than 16°C.
  - The median temperature was 19°C.
  - 75% of the days were cooler than 23°C.

### Key:
- Maximum Temperature (36°C): This was the hottest day recorded, possibly affecting sales patterns.
- Minimum Sales (11 units): The lowest sales occurred on one day, which could be due to external factors like bad weather or low demand.


### 2. Use the Pearson Correlation function to examine the correlation relationship between daily sold units and daily temperature and interpret the result in the Markdown or Text Cell after the code



In [1]:
import scipy.stats

In [22]:
# Extract the two columns
units_sold = df['Daily Units Sold']
temperature = df['Daily Temperature (C)']


In [24]:
type(units_sold)

pandas.core.series.Series

In [23]:
# Calculate Pearson correlation coefficient

correlation_coefficient, p_value = scipy.stats.pearsonr(units_sold, 
                                                        temperature)

In [21]:
print("\n---- Correlation Analysis ----")
print(f"Pearson Correlation Coefficient: {correlation_coefficient:.2f}")
print(f"P-value: {p_value:.4f}")


---- Correlation Analysis ----
Pearson Correlation Coefficient: 0.80
P-value: 0.0000



### Correlation Analysis 
Pearson Correlation Coefficient: 0.80
P-value: 0.0000

The Pearson correlation coefficient of 0.80 indicates a strong positive correlation between the daily units sold and the daily temperature. 


The p-value of 0.0000 (typically displayed as 0.0000 due to rounding) is extremely low, which means the result is statistically significant.

### In other words:

As the temperature goes up, the sales are likely to increase. This might happen because hotter weather could drive demand for products like beverages or other items that help people cool down. Since the correlation is strong and statistically significant, it suggests that temperature plays a significant role in influencing daily sales.

### 3. Sort the data frame according to the number of units sold per day 

In [30]:
df_sorted = df.sort_values(by='Daily Units Sold')
df_sorted

Date  Daily Units Sold  Daily Unit Price  Daily Temperature (C)
219   8/7/2024                11              22.2                     11
220   8/8/2024                11              22.2                     11
234  8/22/2024                11              22.2                     11
223  8/11/2024                19              22.2                     13
226  8/14/2024                19              22.2                     13
..         ...               ...               ...                    ...
28   1/29/2024                95              24.0                     30
30   1/31/2024                96              24.0                     32
27   1/28/2024                97              24.0                     33
16   1/17/2024               100              24.0                     36
4     1/5/2024               100              24.0                     36

[244 rows x 4 columns]

### 4. Write the sorted data frame to a csv file

In [31]:
df_sorted.to_csv('WaterCure_analysis_Hsieh_Luna.csv', index = False )